# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [38]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [39]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4         V5         V6        V7       
1 0.1993901 0.01678375 0.1918293 0.07322714 0.67544436 0.9279782 0.7560318
2 0.9560503 0.56649105 0.7314671 0.67986831 0.04043787 0.3511759 0.4279168
  V8        V9        V10       V11       V12       V13        V14      
1 0.4879994 0.1481411 0.6733738 0.9326594 0.7807921 0.13738306 0.6801351
2 0.9213424 0.7184524 0.9154329 0.7177636 0.3409776 0.06779733 0.2271534
  V15       V16        V17       V18       V19        V20      
1 0.6871657 0.03129947 0.9417735 0.7340596 0.06629425 0.5982321
2 0.1355662 0.90975354 0.7146516 0.1551689 0.42177304 0.2368423

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [40]:
df2 <- df1 %>%
  gather(Key, Value, V1:V20)

head(df2)

Key Value     
1 V1  0.19939010
2 V1  0.95605033
3 V2  0.01678375
4 V2  0.56649105
5 V3  0.19182925
6 V3  0.73146713

## Assignment 2

We will generate some data:

In [41]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [42]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [43]:
df5 <- fulljoin <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook